src/Tools/DynamoShapeManager/Utilities.cs

```diff
@@ -611,11 +622,28 @@ internal static bool IsASMInstallationComplete(IEnumerable<string> filePaths, in
         /// <returns></returns>
         /// <param name="searchPattern">optional - to be used for testing - default is the ASM search pattern</param>
         /// <returns></returns>
-        public static Version GetVersionFromPath(string asmPath, string searchPattern = "ASMAHL*.dll")
+        public static Version GetVersionFromPath(string asmPath, string searchPattern = "*ASMAHL*.*")
         {
             var ASMFilePath = Directory.GetFiles(asmPath, searchPattern, SearchOption.TopDirectoryOnly).FirstOrDefault();
             if (ASMFilePath != null && File.Exists(ASMFilePath))
             {
+#if NET6_0_OR_GREATER
+                if (!OperatingSystem.IsWindows())
+                {
+                    string fileName = Path.GetFileNameWithoutExtension(ASMFilePath);
```

Comment:
I mean, cool, but ugh - I'm surprised theres nothing cross platform here.

Reply:
.so (shared objects) do not have the same metadata as dlls.
I think the recommended versioning technique is to bake the version in the name as a prefix (ex libtsplines.so.10)